Create python-package.yml

tamarl08 opened the pull request:
In this PR I used github template instead of creating a yml file. please review this before [the previous PR](https://github.com/EducationalTestingService/match/pull/17) and if this is fine I will close it.
The tests include python 3.7,3.8,3.9 as opposed to the original testing with only 3.7. The tests fail on python 3.10.

tamarl08 committed:
Create python-package.yml

tamarl08 committed:
Update python-package.yml

add install nose

tamarl08 committed:
Update python-package.yml

update version list

slava92 reviewed (dismissed):
Looks great!

desilinguist commented on .github/workflows/python-package.yml:
```diff
@@ -0,0 +1,35 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
```
We should rename this right since this action is not about packaging?

desilinguist commented:
This looks great, @tamarl08 ! I just had a minor comment. Also, I think we need to resolve at least the branch naming discussion on the discussions [page](https://github.com/EducationalTestingService/match/discussions) before we merge this PR.

tamarl08 committed:
Update and rename python-package.yml to python-test.yml

tamarl08 committed:
Update python-test.yml

add check to develop and stable branches

desilinguist reviewed (dismissed):
LGTM!

desilinguist commented:
Can we get rid of the `.travis.yml` file too please?

tamarl08 committed:
Delete .travis.yml

desilinguist reviewed (dismissed):
Looks great!

desilinguist committed:
Update conda recipe.

- Remove Python 3.7 as a supported version and also prevent Python 3.10
  from being used.
- Use the new host/run combination that works better for noarch packages
  than build/run

desilinguist committed:
Update workflow file.

- Remove Python 3.7.
- Filter branches list to only use develop since the other branches have
  been deleted.
